api/gl/lib/src/gl_shapes.c

```diff
@@ -104,9 +108,18 @@ static void _draw_horizontal_gradient_line(gl_rectangle_t *rect, gl_rectangle_t*
 static void _draw_vertical_gradient_line(gl_rectangle_t *rect, gl_rectangle_t* border_rect)
 {
     gl_color_t brush_color;
+    gl_rectangle_t my_rect;
+
+    my_rect.height = 1;
+    my_rect.width = rect->width;
+    my_rect.top_left.y = rect->top_left.y + rect->height;
+    my_rect.top_left.x = rect->top_left.x;
 
-    brush_color = gl_gradient(instance.gradient_color.from, instance.gradient_color.to, (rect->top_left.y - border_rect->top_left.y) / (float) border_rect->height);
-    instance.driver.fill_f(rect, brush_color);
+    while (--my_rect.top_left.y >= rect->top_left.y)
+    {
+        brush_color = gl_gradient(instance.gradient_color.from, instance.gradient_color.to, (my_rect.top_left.y - border_rect->top_left.y) / (float) border_rect->height);
+        instance.driver.fill_f(&my_rect, brush_color);
+    }
```

Comment:
Why change this API?

It doesn't look like it's needed.

Please explain why it's been added or remove the additional code.

Reply:
It is necessary to change this API in this way, because then I can just call `_draw_vertical_gradient_line()` with rectangle of height > 1 and have it painted in a gradient way. If I wanted that with the old code I would need a loop iterating over the rectangle height in every place where I call that function i.e. what the internal loop of this function does.

Here is the comparison of the result of the same ellipse drawing algorithm with old and new `_draw_vertical_gradient_line()` definition, respectively: <https://imgur.com/a/9zUfvjQ>

So this change is generally useful. It is obviously modeled on the definition of `_draw_horizontal_gradient_line()`. It should not affect already existing code.

Reply:
I see. Good thing you took that into consideration as well.

Ok to stay then.